### PR TITLE
DOC: Document that read_hdf can use pickle

### PR DESCRIPTION
**Pull request by TomAugspurger**
This documents that `read_hdf`, which uses PyTables, might invoke pickle to deserialize arrays that were serialize with pickle.

PyTables clearly documents this at http://www.pytables.org/usersguide/libref/declarative_classes.html#tables.ObjectAtom, but pandas users may not realize that if they aren't aware that pandas uses pytables under the hood.

I've tried to include the warning in all the methods that eventually read data, but it's a bit hard to say if I've gotten them all.

Ideally we would also like to provide an `allow_pickle=False` keyword, but that would be best implemented in PyTables, and used by us. I opened https://github.com/PyTables/PyTables/issues/813 for discussion.

### CODE REVIEW

File: doc/source/user_guide/io.rst [hunk 1]
```diff
@@ -3441,10 +3441,11 @@ for some advanced strategies
 
 .. warning::
 
-   pandas requires ``PyTables`` >= 3.0.0.
-   There is a indexing bug in ``PyTables`` < 3.2 which may appear when querying stores using an index.
-   If you see a subset of results being returned, upgrade to ``PyTables`` >= 3.2.
-   Stores created previously will need to be rewritten using the updated version.
+   Pandas uses PyTables for reading and writing HDF5 files, which allows
```
Comment (jreback) [inline doc/source/user_guide/io.rst:3444]: note that this only applies to fixed stores; tables do not use pickle

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by WillAyd** [comment]
lgtm ex Jeff's comments

**Comment by TomAugspurger** [comment]
FYI, I tagged this for 1.1.1. Wouldn't normally backport doc-only changes but this is tangentially security related so I think it's better to backport.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @TomAugspurger

**Pull request merged**